### ISSUE TITLE
Fix `at-rule-no-unknown` false negatives for `@function`

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#8843]
### What minimal example or steps are needed to reproduce the bug?

```css
@function --foo() {}
```

### What minimal configuration is needed to reproduce the bug?

```
{
  "rules": {
    "at-rule-no-unknown": true
  }
}
```

### How did you run Stylelint?

[demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAzArgOzAFwJYQwAIBaYlCCACgEpDgBfEAGhBVwBsYA5AQwFs4iGAA9+AB04A6MAGcZzcATYBzBCBFiIAJ2yFYKHmna7gAHQynsIqxigz4hANoByGdgCendrgzZikDBViNx5bHi0oZwBdJnN6AG5zBQCVADFtPh5sNTdPGG9faSVcZUk+ACt5FlgxeUQzIkJTEFyvH2xmh2b2LJg3ZtjG5tb89v9i5WDsUKhwqE6mkB6rfpA4kHogA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest"
}
```

### Resources

support : https://caniuse.com/wf-function
spec: https://drafts.csswg.org/css-mixins-1/#function-rule
related: #1612

### What actually happened?

```
Unexpected unknown at-rule "@function" (at-rule-no-unknown)
```

### Do you have a proposal to fix the bug?

update https://github.com/stylelint/stylelint/blob/main/lib/reference/atKeywords.mjs
